syntax = "proto3";
package tool;
import "google/protobuf/struct.proto";

message WebSearchResult {
  string title = 1;
  string url = 2;
  string snippet = 3;
}

message ToolRequest {
  string tool_name = 1;
  google.protobuf.Struct params = 2;
}

message ToolResponse {
  bool success = 1;
  string message = 2;
  repeated WebSearchResult results = 3;
  map<string, string> metadata = 4;
}

service ToolService {
  rpc CallTool(ToolRequest) returns (ToolResponse);
}